NUnitLite/TouchRunner/MacRunner.cs

```diff
@@ -21,12 +22,31 @@ public static async Task<int> MainAsync (IList<string> arguments, bool requiresN
 
 			if (requiresNSApplicationRun) {
 				var app = NSApplication.SharedApplication;
+				// This window will let us stop the NSApplication's run loop after the test run
 				app.InvokeOnMainThread (async () => {
+					var rect = new CGRect (0, 0, 200, 50);
+					var window = new NSWindow (rect, NSWindowStyle.Titled | NSWindowStyle.Closable | NSWindowStyle.Miniaturizable | NSWindowStyle.Resizable, NSBackingStore.Buffered, false);
+					window.Title = "Running tests...";
+					window.MakeKeyAndOrderFront (app);
+
+					// run the tests
 					success = await RunTestsAsync (options, assemblies);
-					// The only reliable way to stop NSApplication.Run is to call NSApplication.Terminate, which will
-					// terminate the app, but won't allow us to specify the exit code. So we need an callback that will
-					// exit the process
-					exitProcess (success ? 0 : 1);
+
+					app.BeginInvokeOnMainThread (() => {
+						// First try to stop the app nicely. This only works if the app is showing a window.
+						app.Stop (app);
+						// Stopping the run loop only works something else is processed by the run loop, so add an event.
```

Comment:
Nit:
`// Stopping the run loop only works something` ->`// Stopping the run loop only works if something`